inc/core/vt_defs.h

```diff
@@ -10,20 +10,31 @@
 #include <stdint.h>
 #include <stdio.h>
 
-#define abs_custom(x) ((x) > 0 ? (x) : -(x))
+#define abs_custom(x) ((x) > 0.0f ? (x) : -(x))
```

Comment:
this change would result in anyone using abs_custom to be forced to use use floating point, even if the type is int.

Also, why is there a custom version here instead of using abs/fabs/fabsf?

Reply:
had warnings converted to errors due to double promotion check being enforced

Reply:
Use "fabsf" from <math.h>, this is the float version so you wont get the errors. [https://en.cppreference.com/w/c/numeric/math/fabs](https://en.cppreference.com/w/c/numeric/math/fabs)
I tested this in it resulted in a small reduction in FLASH space. Just need to test that there isnt a regressions as a result of switching this